ambari-common/src/main/python/resource_management/libraries/execution_command/execution_command.py

```diff
@@ -54,32 +54,130 @@ def __get_value(self, key, default_value=None):
     except:
       return default_value
 
+  """
+  Global variables section
+  """
+
+  def get_module_configs(self):
+    return self.__module_configs
+
   def get_module_name(self):
     return self.__get_value("serviceName")
 
-  def get_stack_name(self):
-    return self.__get_value("hostLevelParams/stack_name")
+  def get_component_type(self):
+    return self.__get_value("role")
+
+  def get_component_instance_name(self):
+    """
+    At this time it returns hardcoded 'default' name
+    :return:
+    """
+    return self.__get_value("default")
+
+  def get_servicegroup_name(self):
+    return self.__get_value("serviceGroupName")
+
+  def get_cluster_name(self):
+    return self.__get_value("clusterName")
+
+  """
+  Ambari variables section
+  """
+
+  def get_jdk_location(self):
+    return self.__get_value("ambariLevelParams/jdk_location")
+
+  def get_jdk_name(self):
+    return self.__get_value("ambariLevelParams/jdk_name")
+
+  def get_java_home(self):
+    return self.__get_value("ambariLevelParams/java_home")
+
+  def get_java_version(self):
+    return int(self.__get_value("ambariLevelParams/java_version"))
+
+  def get_jce_name(self):
+    return self.__get_value("ambariLevelParams/jce_name")
+
+  def get_db_driver_file_name(self):
+    return self.__get_value('ambariLevelParams/db_driver_filename')
+
+  def get_db_name(self):
```

Comment:
We still need to handle default value "None" for all the params.

Reply:
The default is None, if no key is found, please see:

  def __get_value(self, key, default_value=None):
    """
    A private method to query value with the full path of key, if key does not exist, return default_value
    :param key:
    :param default_value:
    :return:
    """
    sub_keys = key.split('/')
    value = self.__execution_command
    try:
      for sub_key in sub_keys:
        value = value[sub_key]
      return value
    except:
      return default_value